src/borg/archiver.py

```diff
@@ -652,10 +709,10 @@ def _process(self, *, path, parent_fd=None, name=None,
                 # could trigger an error, e.g. if access is forbidden, see #3209.
                 if not matcher.recurse_dir:
                     return
+                recurse_excluded_dir = True
                 with backup_io('stat'):
                     st = os_stat(path=path, parent_fd=parent_fd, name=name, follow_symlinks=False)
-                recurse_excluded_dir = stat.S_ISDIR(st.st_mode)
-                if not recurse_excluded_dir:
+                if not stat.S_ISDIR(st.st_mode):
                     return
```

Comment:
note for other potential reviewers: this is just a style fix, functionality is the same.